dtfs-central-api/api-tests/helpers/with-validate-audit-details.api-tests.ts

```diff
@@ -8,50 +8,46 @@ import { MOCK_TFM_USER } from '../mocks/test-users/mock-tfm-user';
   payloadWithoutAuditDetails: T & { auditDetails?: AuditDetails };
   validUserTypes: ('tfm' | 'portal' | 'system')[];
-  successfulStatus?: number;
 };
 
-export const withValidateAuditDetailsTests = <T extends object>({
-  makeRequest,
-  payloadWithoutAuditDetails,
-  validUserTypes,
-  successfulStatus = 200,  
-}: Params<T>) => {
-  it('should return 400 if no auditDetails provided', async () => {
-    const { status, body } = await makeRequest({ ...payloadWithoutAuditDetails });
+export const withValidateAuditDetailsTests = <T extends object>({ makeRequest, validUserTypes }: Params<T>) => {
+  describe('when validating audit details', () => {
+    it('should return 400 if no auditDetails provided', async () => {
+      const { status, body } = await makeRequest();
 
-    expect(body).toEqual({ status: 400, message: 'Invalid auditDetails, Missing property `userType`' });
-    expect(status).toBe(400);
-  });
+      expect(body).toEqual({ status: 400, message: 'Invalid auditDetails, Missing property `userType`' });
+      expect(status).toBe(400);
+    });
 
-  const testCases = [
-    {
-      auditDetails: generateSystemAuditDetails(),
-      expectedStatus: validUserTypes.includes('system') ? successfulStatus : 400,
-    },
-    {
-      auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
-      expectedStatus: validUserTypes.includes('portal') ? successfulStatus : 400,
-    },
-    {
-      auditDetails: generateTfmAuditDetails(MOCK_TFM_USER._id),
-      expectedStatus: validUserTypes.includes('tfm') ? successfulStatus : 400,
-    },
-  ];
+    const testCases = [
+      {
+        auditDetails: generateSystemAuditDetails(),
+        expectedStatus: validUserTypes.includes('system') ? 200 : 400,
+      },
+      {
+        auditDetails: generatePortalAuditDetails(MOCK_PORTAL_USER._id),
+        expectedStatus: validUserTypes.includes('portal') ? 200 : 400,
+      },
+      {
+        auditDetails: generateTfmAuditDetails(MOCK_TFM_USER._id),
+        expectedStatus: validUserTypes.includes('tfm') ? 200 : 400,
+      },
+    ];
 
-  it.each(testCases)('it should have status $expectedStatus if the userType is $auditDetails.usertype', async ({ auditDetails, expectedStatus }) => {
-    const { status, body } = await makeRequest({ ...payloadWithoutAuditDetails, auditDetails });
+    it.each(testCases)('it should have status $expectedStatus if the userType is $auditDetails.usertype', async ({ auditDetails, expectedStatus }) => {
+      const { status, body } = await makeRequest(auditDetails);
 
-    expect(status).toBe(expectedStatus);
-    if (expectedStatus === 400) {
-      expect(body).toEqual({
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-        message: expect.stringContaining('Invalid auditDetails, userType must be'),
-        status: 400,
-      });
-    }
+      expect(status).toBe(expectedStatus);
+      if (expectedStatus === 400) {
```

Comment:
I'm not the biggest fan of having an if statement here, thoughts?

Reply:
me neither -- bit of a code smell. A way around this would be construct two test cases lists -- one for 2xxs one for 4xxs, and then have two different `it('bla')` statements